core/config_main.go

```diff
@@ -0,0 +1,490 @@
+package core
+
+import (
+	"errors"
+	"fmt"
+	"io/fs"
+	"os"
+	"path/filepath"
+	"sort"
+	"strings"
+
+	"github.com/Masterminds/semver/v3"
+	"gopkg.in/yaml.v3"
+)
+
+const (
+	configEnvVar        = "CHANGIE_CONFIG_PATH"
+	timeFormat   string = "20060102-150405"
+
+	CreateFileMode os.FileMode = 0644
+	CreateDirMode  os.FileMode = 0755
+
+	AutoLevel  = "auto"
+	MajorLevel = "major"
+	MinorLevel = "minor"
+	PatchLevel = "patch"
+	NoneLevel  = "none"
+	EmptyLevel = ""
+)
+
+var ConfigPaths []string = []string{
+	".changie.yaml",
+	".changie.yml",
+}
+
+var ErrConfigNotFound = errors.New("no changie config found")
+
+// GetVersions will return, in semver sorted order, all released versions
+type GetVersions func(Config) ([]*semver.Version, error)
+
+// Config handles configuration for a project.
+//
+// Custom configuration path:
+//
+// By default Changie will try and load ".changie.yaml" and ".changie.yml" before running commands.
+// If you want to change this path set the environment variable `CHANGIE_CONFIG_PATH` to the desired file.
+//
+// `export CHANGIE_CONFIG_PATH=./tools/changie.yaml`
+//
+// [Template Cache](#templatecache-type) handles all the template executions.
+type Config struct {
+	// Which configuration version we are using. Version 1 is used if none is provided.
+	Version int `yaml:"version"`
+
+	// Directory for change files, header file and unreleased files.
+	// Relative to project root.
+	// example: yaml
+	// changesDir: .changes
+	RootDir string `yaml:"rootDir" default:".changes"`
+
+	// Prefix of environment variables to load for templates.
+	// The prefix is removed from resulting key map.
+	// example: yaml
+	// # if we have an environment variable like so:
+	// # export CHANGIE_PROJECT=changie
+	// # we can use that in our templates if we set the prefix
+	// envPrefix: "CHANGIE_"
+	// versionFormat: "New release for {{.Env.PROJECT}}"
+	EnvPrefix string `yaml:"envPrefix,omitempty"`
+
+	// TODO: docs
+	Fragment     FragmentConfig     `yaml:"fragment"`
+	Project      ProjectConfig      `yaml:"project"`
+	Component    ComponentConfig    `yaml:"component"`
+	Kind         KindConfig         `yaml:"kind"`
+	Change       ChangeConfig       `yaml:"change"`
+	ReleaseNotes ReleaseNotesConfig `yaml:"releaseNotes"`
+	Changelog    ChangelogConfig    `yaml:"changelog"`
```

Comment:
Consolidated into 1, based on current implementation 

```suggestion
	ReleaseNotes ReleaseNotesConfig `yaml:"releaseNotes"`
```
Or alternatively if implementation switched
```suggestion
	Changelog    ChangelogConfig    `yaml:"changelog"`
```